BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
+{
+    [JsonPropertyName("currencyCode")] public string CurrencyCode { get; set; }
+    [JsonPropertyName("rate")] public float RateValue { get; set; }
+}
+
+internal class ExchangeRateService
+{
+    private const string MainCurrenciesUrl = "https://api.cnb.cz/cnbapi/exrates/daily";
+
+    public IEnumerable<Rate> GetExchangeRates(List<string> currencies)
+    {
+        HttpClient client = new HttpClient();
+			
+        var data = client.GetStringAsync(MainCurrenciesUrl).Result;
```

Comment:
When using an async method, always use `await` to get the result from it instead of `.Result` or any other method:

`var data = await client.GetStringAsync(MainCurrenciesUrl);`

Using `await` requires the method calling it to be marked as `async`, and to wrap the return type in `Task<T>`. We also add `Async` to the end of the method name, to clarify for any caller that they are calling an async method:

`public async Task<IEnumerable<Rate>> GetExchangeRatesAsync(List<string> currencies)`

Not using `await` when calling async methods should generally only be done as a last resort, as it can hamper performance and, more importantly, introduce deadlock scenarios to a codebase.